osu.Game/Overlays/MusicController.cs

```diff
@@ -307,13 +304,36 @@ private void beatmapChanged(ValueChangedEvent<WorkingBeatmap> beatmap)
             }
 
             current = beatmap.NewValue;
+
+            if (CurrentTrack.IsDummyDevice || !beatmap.OldValue.BeatmapInfo.AudioEquals(current?.BeatmapInfo))
+                changeTrack();
+
             TrackChanged?.Invoke(current, direction);
 
             ResetTrackAdjustments();
 
             queuedDirection = null;
         }
 
+        private void changeTrack()
+        {
+            CurrentTrack.Expire();
+            CurrentTrack = new DrawableTrack(current.GetTrack());
+            CurrentTrack.Completed += () => onTrackCompleted(current);
+
+            AddInternal(CurrentTrack);
```

Comment:
As mentioned by @frenzibyte this is causing crashes in test scenes - as far as I'm aware value changed callbacks run on the thread that changed the value, which is not necessarily the update thread, so it seems this needs a schedule.

A normal schedule still doesn't always work (crashes i.e. in `TestSceneNightcoreBeatContainer`); `Scheduler.AddOnce()` seems to not crash any of the test scenes I checked but I'm unsure whether it's a correct thing to do, or if it should be a normal schedule and all crashing sites should apply local fixes.

Reply:
This maaaay be fine. Will have to check it behaves correctly with `restartTrack()`.